xcat-inventory/xcclient/inventory/manager.py

```
@@ -667,6 +662,24 @@ def importobj(srcfile,srcdir,objtype,objnames=None,dryrun=None,version=None,upda
      if objnames:
          objnamelist=[n.strip() for n in objnames.split(',')]
 
+     vardict = {}
+
+     if env_files:
+        for env_file in env_files:
+            try:
+                f = open(env_file)
+                vardict.update( yaml.load(f) )
+            except Exception,e:
+                raise InvalidFileException("Error: Failed to load yaml file '%s', please check ..." % env_file)
+            f.close()
```

`f.close()` should better be moved to before `raise InvalidFileException...`